Core/Config.cpp

```diff
@@ -601,6 +586,7 @@ static ConfigSetting generalSettings[] = {
 	ConfigSetting("EnablePlugins", &g_Config.bLoadPlugins, true, true, true),
 
 	ReportedConfigSetting("IgnoreCompatSettings", &g_Config.sIgnoreCompatSettings, "", true, true),
+	ConfigSetting("SettingsVersion", &g_Config.uSettingsVersion, 0u, true, true), // Per game for game configs
```

Comment:
We traditionally haven't had a version number for settings. I hope the addition of this will not lead to a proliferation of changes, because testing all kinds of settings upgrades will be very laborious...

Reply:
We could have a bool specific for this port like `bRightAnalogSettingPorted` that would reduce the temptations to add new settings version.

Anyway I see your concern with this PR, if/when you decide what's best for the project I'll change and rebase accordingly :)

Reply:
I'm taking back my comment here, SettingsVersion is the better option, otherwise if we make more corrections in the future, we'll just end up with more specific bools and the combinations get harder to understand.